app/javascript/components/shared/CustomerForm.jsx

```diff
@@ -84,22 +84,30 @@ export const CustomerForm = () => {
 
   const handleEmailChange = (e) => {
     const { value } = e.target
-
     setCustomer((prev) => ({ ...prev, email: value }))
+
+    // Only validate if there's an existing email error to clear it immediately
+    if (errors.email) {
+      if (value && !INPUT_VALIDATORS.emailFormat.test(value)) {
+        setErrors((prev) => ({ ...prev, email: 'Invalid email format' }))
+        setIsNextDisabled(true)
+      } else {
+        setErrors((prev) => ({ ...prev, email: '' }))
+        setIsNextDisabled(!customer.company_name || (value && !INPUT_VALIDATORS.emailFormat.test(value)))
+      }
+    }
```

Comment:
Instead of so much nesting of conditionals maybe add at least the first one as a guard claus? e.g.
```
if (!errors.email) return;
```